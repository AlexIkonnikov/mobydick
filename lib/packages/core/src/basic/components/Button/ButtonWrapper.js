"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = require("react");
const react_native_1 = require("react-native");
const ButtonWrapper = (0, react_1.forwardRef)((props, ref) => {
    return <react_native_1.Button ref={ref} {...props}/>;
});
exports.default = ButtonWrapper;
