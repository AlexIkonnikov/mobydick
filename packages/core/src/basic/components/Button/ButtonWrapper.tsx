import {forwardRef} from 'react';
import {Button as DefaultButton} from 'react-native';

import {IButtonWrapperProps, IButtonWrapper} from './types';

const ButtonWrapper = forwardRef<IButtonWrapper, IButtonWrapperProps>(
  (props, ref) => {
    return <DefaultButton ref={ref} {...props} />;
  },
);
export default ButtonWrapper;
