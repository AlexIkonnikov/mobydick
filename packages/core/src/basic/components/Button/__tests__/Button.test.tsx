import React from 'react';
import {render} from '@testing-library/react-native';

import ButtonWrapper from '../ButtonWrapper';

describe('@lad-tech/mobydick-core/Button', () => {
  it('renders correctly', () => {
    const {toJSON} = render(<ButtonWrapper title={'button'} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
