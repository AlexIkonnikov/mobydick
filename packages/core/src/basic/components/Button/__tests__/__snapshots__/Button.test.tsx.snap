// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-core/Button renders correctly 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {},
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      button
    </Text>
  </View>
</View>
`;
